icu4c/source/tools/toolutil/toolutil.cpp

```diff
@@ -166,15 +166,11 @@ findBasename(const char *filename) {
     const char *basename=uprv_strrchr(filename, U_FILE_SEP_CHAR);
 
 #if U_FILE_ALT_SEP_CHAR!=U_FILE_SEP_CHAR
-#if !(U_PLATFORM == U_PF_CYGWIN && U_PLATFORM_USES_ONLY_WIN32_API)
-    if(basename==NULL)
-#endif
-    {
-        /* Use lenient matching on Windows, which can accept either \ or /
-           This is useful for environments like Win32+CygWin which have both.
-        */
-        basename=uprv_strrchr(filename, U_FILE_ALT_SEP_CHAR);
-    }
+    //be lenient about pathname separators on Windows, like official implementation of C++17 std::filesystem in MSVC
+    //would be convinient to merge this loop with the one above, but alas, there is no such solution in the standard library
+    const char *alt_basename=uprv_strrchr(filename, U_FILE_ALT_SEP_CHAR);
+    if(alt_basename>basename)
```

Comment:
```suggestion
    if(alt_basename>basename) {
        basename=alt_basename;
     }
```

The code in this file doesn't really follow the ICU style guides... but would you mind using the parenthesis for the if statement block though? (The ICU style guide says to always enclose the block anyways.)

Reply:
How do you suggest changing the suggestion? I am confused.
You forgot to include the line below and now it has a duplicate of `basename=alt_basename;`

Reply:
Sorry for the confusion... I think the GitHub suggestion got messed up as I started the comment by only commenting on one line.
Your suggested change here https://github.com/unicode-org/icu/pull/1398#discussion_r501288905 looks good. :)

Reply:
I squashed your suggestions into my initial commit^